kibana-reports/server/routes/utils/visualReportHelper.ts

```diff
@@ -18,85 +18,131 @@ import { FORMAT } from './constants';
-      height: windowHeight,
-    });
-
-    // TODO: this element is for Dashboard page, need to think about addition params to select html element with source(Visualization, Dashboard)
-
-    const timeCreated = new Date().toISOString();
-    const fileName = getFileName(itemName, timeCreated) + '.' + FORMAT.png;
-
-    const buffer = await page.screenshot({
-      fullPage: true,
-    });
-
-    //TODO: Add header and footer, phase 2
-
-    await browser.close();
-    return { timeCreated, dataUrl: buffer.toString('base64'), fileName };
+    return renderReport(
+      'image/png',
+      url,
+      source,
+      itemName,
+      windowWidth,
+      windowHeight,
+      header,
+      footer
+    );
   } catch (error) {
     throw error;
   }
 };
 
 export const generatePDF = async (
   url: string,
+  source: string,
   itemName: string,
   windowWidth: number,
-  windowHeight: number
+  windowHeight: number,
+  header: string,
+  footer: string
 ): Promise<{ timeCreated: string; dataUrl: string; fileName: string }> => {
   try {
-    const browser = await puppeteer.launch({
-      headless: true,
-    });
-    const page = await browser.newPage();
+    return renderReport(
+      'application/pdf',
+      url,
+      source,
+      itemName,
+      windowWidth,
+      windowHeight,
+      header,
+      footer
+    );
+  } catch (error) {
+    throw error;
+  }
+};
 
-    await page.setDefaultNavigationTimeout(0);
-    await page.setViewport({
-      width: windowWidth,
-      height: windowHeight,
-    });
+async function renderReport(
+  format: string,
+  url: string,
+  source: string,
+  itemName: string,
+  windowWidth: number,
+  windowHeight: number,
+  header: string,
+  footer: string
+) {
+  if (source === 'Dashboard') {
+    windowWidth = 1440;
+    windowHeight = 2560;
+  } else if (source === 'Visualization') {
+    windowWidth = 900;
+    windowHeight = 1300;
+  }
 
-    await page.goto(url, { waitUntil: 'networkidle0' });
+  const browser = await puppeteer.launch({
+    headless: true,
+  });
+  const page = await browser.newPage();
 
-    const timeCreated = new Date().toISOString();
-    const fileName = getFileName(itemName, timeCreated) + '.' + FORMAT.pdf;
-    // The scrollHeight value is equal to the minimum height the element would require in order to fit
-    // all the content in the viewport without using a vertical scrollbar
-    const scrollHeight = await page.evaluate(
-      () => document.documentElement.scrollHeight
-    );
+  await page.setDefaultNavigationTimeout(0);
+  await page.setViewport({
+    width: windowWidth,
+    height: windowHeight,
+  });
+
+  await page.goto(url, { waitUntil: 'networkidle0' });
+  let element;
+
+  if (source === 'Dashboard') {
+    await page.waitForSelector('#dashboardViewport');
+    element = await page.$('#dashboardViewport');
+  } else if (source === 'Visualization') {
+    await page.waitForSelector('.visChart');
+    element = await page.$('.visChart');
```

Comment:
https://www.elastic.co/guide/en/kibana/current/reporting-integration.html#_screenshot_capturing_attributes
See if you can use those attributes to capture the screen. Also, leave some comments here to describe why you use certain elements in the html for screen capture.

Reply:
The implementation is different and will require an implementation change in the server-side for the screen capture. Will look into it but if it requires a lot of bandwidth may flag as enhancement